res/controllers/Denon-MC7000-scripts.js

```diff
@@ -573,34 +610,154 @@ MC7000.PadButtons = function(channel, control, value, status, group) {
                     if (MC7000.prevSamplerStop) {
-                        // stop playing all other samplers ...
-                        for (j = 1; j <=8; j++) {
-                            engine.setValue("[Sampler" + j + "]", "cue_gotoandstop", 1);
+                        // stop playing all other samplers on this deck
+                        for (j = 1; j <= 8; j++) {
+                            samplerOffsetJ = deckOffset * 8 + j;
+                            if (engine.getValue("[Sampler" + samplerOffsetJ + "]", "play") === 1) {  // if sampler is playing then stop it
+                                engine.setValue("[Sampler" + samplerOffsetJ + "]", "cue_gotoandstop", 1);
+                                midi.sendShortMsg(0x94 + deckOffset, 0x14 + j - 1, MC7000.padColor.samplerloaded);
+                            }
                         }
                     }
                     // ... before the actual sampler to play gets started
-                    engine.setValue("[Sampler" + i + "]", "cue_gotoandplay", 1);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "pregain", 1);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "cue_gotoandplay", 1);
+                    midi.sendShortMsg(0x94 + deckOffset, 0x14 + i - 1, MC7000.padColor.samplerplay);
+                }
+            } else if (control === 0x1C + i - 1 && value >= 0x01) { //shifted button deactivates playing sampler or ejects sampler
+                if (MC7000.SamplerQty === 16) {
+                    if ((deckOffset) >= 2) {
+                        deckOffset = deckOffset + 2;
+                    }
                 }
-            } else if (control === 0x1C + i - 1 && value >= 0x01) {
-                if (engine.getValue("[Sampler" + i + "]", "play") === 1) {
-                    engine.setValue("[Sampler" + i + "]", "cue_gotoandstop", 1);
+                if (engine.getValue("[Sampler" + samplerOffset + "]", "play") === 1) {
+                    engine.setValue("[Sampler" + samplerOffset + "]", "cue_gotoandstop", 1);
                     midi.sendShortMsg(0x94 + deckOffset, 0x1C + i - 1, MC7000.padColor.samplerloaded);
                 } else {
-                    engine.setValue("[Sampler" + i + "]", "eject", 1);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "eject", 1);
                     midi.sendShortMsg(0x94 + deckOffset, 0x1C + i - 1, MC7000.padColor.sampleroff);
-                    engine.setValue("[Sampler" + i + "]", "eject", 0);
+                    engine.setValue("[Sampler" + samplerOffset + "]", "eject", 0);
+                }
+            }
+        }
+    } else if (MC7000.PADModeVelSamp[deckOffset]) {
+        samplerOffset = 0;
+        samplerOffsetJ = 0;
+        for (i = 1; i <= 8; i++) {
+            if (MC7000.SamplerQty === 16) {
+                if (deckOffset >= 2) {
+                    deckOffset = deckOffset - 2;
```

Comment:
This code needs a comment, that explains, what is done here

Reply:
In case the user uses 16 samplers, 8 on the left deck 1 and 8 on the right deck 2, When switching to deck 3 the same 8 samplers as used for deck 1 have to be used on deck 3 (and the same samplers for deck 4 than for deck 2). 

Do you want a comment inside the code or is explanation here sufficient?

Reply:
done